File: tinyredis-core/src/main/java/com/github/cjqcn/tinyredis/core/db/impl/RedisDbImpl.java
package com.github.cjqcn.tinyredis.core.db.impl;

import com.github.cjqcn.tinyredis.core.db.Dict;
import com.github.cjqcn.tinyredis.core.db.RedisDb;

public class RedisDbImpl implements RedisDb {
    private final int id;
    private Dict dict = new BaseDict();
    private ExpiresDict expires = new ExpiresDict();

    public RedisDbImpl(final int id) {
        this.id = id;
    }

    @Override
    public int id() {
        return id;
    }

    @Override
    public Dict dict() {
        return dict;
    }

    @Override
    public ExpiresDict expires() {
        return expires;
    }
}

File: tinyredis-core/src/main/java/com/github/cjqcn/tinyredis/core/db/impl/RedisDbImpl.kt
package com.github.cjqcn.tinyredis.core.db.impl

import com.github.cjqcn.tinyredis.core.db.Dict
import com.github.cjqcn.tinyredis.core.db.RedisDb

class RedisDbImpl : RedisDb {

    val dict = BaseDict();
    val expires = ExpiresDict();

    override fun id(): Int {
        return 1
    }

    override fun dict(): Dict {
        return dict
    }

    override fun expires(): Dict {
        return expires
    }

    override fun avgTtl(): Long {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}